File: courier_application/app/src/main/java/by/bstu/vs/stpms/courier_application/model/database/model/entity/Order.java
package by.bstu.vs.stpms.courier_application.model.database.model.entity;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

import java.sql.Timestamp;
import java.util.Collection;

import lombok.Data;

@Data
@Entity(
        tableName = "order",
        foreignKeys = {
                @ForeignKey(entity = Customer.class, parentColumns = "id", childColumns = "customerId"),
                @ForeignKey(entity = User.class, parentColumns = "id", childColumns = "courierId")
        }
)
public class Order {
    @PrimaryKey
    private long id;
    private String address;
    private String info;
    private String state;
    private Long customerId;
    private Long courierId;
    private Collection<Long> orderedsId;
    private Timestamp orderedAt;
    private Timestamp deliveredAt;
}

File: courier_application/app/src/main/java/by/bstu/vs/stpms/courier_application/model/entity/Order.kt
package by.bstu.vs.stpms.courier_application.model.entity

import java.sql.Timestamp

class Order (
     val address: String,
     val info: String? = null,
     val state: String,
     val customer: Customer,
     val courierId: Long? = null,
     val ordered: Collection<Ordered>,
     val orderedAt: Timestamp? = null,
     val deliveredAt: Timestamp? = null
) : AbstractEntity()